modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/ChainedIterator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.util;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Iterator, that chains several iterator into a single one.
+ *
+ * @param <T> A type of the item.
+ */
+public class ChainedIterator<T> implements Iterator<T> {
```

Comment:
We already have similar structure.
`org.apache.ignite.internal.util.CollectionUtils.concat(Iterator ..)`
This can be replaced with utility method call.

Reply:
I knew we have something similar, but was unable to find it... Thanks, replaced with utility method